sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosDataWriteFactory.scala

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.spark
+
+import java.util.UUID
+
+import com.azure.cosmos.implementation.TestConfigurations
+import com.azure.cosmos.{ConsistencyLevel, CosmosClientBuilder}
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.connector.write.{DataWriter, DataWriterFactory, WriterCommitMessage}
+import org.apache.spark.sql.types.{IntegerType, StringType, StructField, StructType}
+
+class CosmosDataWriteFactory extends DataWriterFactory with CosmosLoggingTrait {
+  logInfo(s"Instantiated ${this.getClass.getSimpleName}")
+
+  override def createWriter(i: Int, l: Long): DataWriter[InternalRow] = new CosmosWriter()
+
+  class CosmosWriter() extends DataWriter[InternalRow] {
+    logInfo(s"Instantiated ${this.getClass.getSimpleName}")
+
+    // TODO moderakh account config and databaseName, containerName need to passed down from the user
+    val client = new CosmosClientBuilder()
+      .key(TestConfigurations.MASTER_KEY)
```

Comment:
I assuem long term we would want to have a cache similar to what I added in the 3.* release for today's OLTP connector? If so I can take a stab at that early next week.

Reply:
Similar to CosmosDBConnectionCache.scala